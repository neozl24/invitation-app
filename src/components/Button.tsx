interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode
}

export const Button = ({
  className = '',
  children,
  ...props
}: ButtonProps) => {
  const baseClassName = `
    flex
    items-center
    h-[50px]
    mb-20
    px-4
    border
    text-sm
    text-neutral-400
    border-neutral-600
    hover:bg-neutral-100
    active:hover:bg-neutral-50
  `
  return (
    <button className={`${baseClassName} ${className}`} {...props}>
      {children}
    </button>
  )
}