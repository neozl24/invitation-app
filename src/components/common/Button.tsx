interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode
}

export const Button = ({
  className = '',
  children,
  disabled,
  ...props
}: ButtonProps) => {
  const baseClassName = `
    flex
    items-center
    justify-center
    h-[50px]
    px-4
    border
    text-sm
    outline-none
  `
  const statusClassName = disabled ? `
    text-neutral-300
    border-neutral-300
  ` : `
    text-neutral-500
    border-neutral-600
    focus-visible:outline-blue-700
    focus-visible:outline-2
    hover:bg-neutral-100
    active:hover:bg-neutral-50
  `

  return (
    <button
      className={`${baseClassName} ${statusClassName} ${className}`}
      disabled={disabled}
      {...props}
    >
      {children}
    </button>
  )
}