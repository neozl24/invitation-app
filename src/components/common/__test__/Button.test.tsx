import { render, screen, fireEvent } from '@testing-library/react'
import { Button } from '../Button'

test('renders children as content and can trigger callback when clicked', () => {
  const { asFragment, getByText } = render(<Button>Click Me</Button>)

  expect(asFragment()).toMatchSnapshot()
  expect(getByText('Click Me')).toBeDefined()
})

test('Can trigger callback when clicked', () => {
  const callback = vi.fn()

  render(<Button onClick={callback}>Click Me</Button>)

  const button = screen.getByRole('button', { name: 'Click Me' })
  expect(button).toBeInTheDocument()

  fireEvent.click(button)
  expect(callback).toHaveBeenCalled()
})
