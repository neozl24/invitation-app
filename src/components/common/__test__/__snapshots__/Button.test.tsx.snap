// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders children as content and can trigger callback when clicked 1`] = `
<DocumentFragment>
  <button
    class="
    flex
    items-center
    justify-center
    h-[50px]
    px-4
    border
    text-sm
    font-medium
    outline-none
   
    text-neutral-500
    border-neutral-600
    focus-visible:outline-blue-700
    focus-visible:outline-2
    hover:bg-neutral-100
    active:hover:bg-neutral-50
   "
  >
    Click Me
  </button>
</DocumentFragment>
`;
